sidekick/views.py

```diff
@@ -68,3 +70,40 @@ def send_wa_template_message(request):
     )
 
     return JsonResponse(json.loads(result.content), status=result.status_code)
+
+
+class CheckContactView(APIView):
+    """
+    Accepts Org id and msisdn
+    Checks the Turn API to see if the contact is valid
+    Returns a JsonResponse containing status as valid/invalid
+    """
+
+    def get(self, request, org_id, msisdn, *args, **kwargs):
+        try:
+            org = Organization.objects.get(id=org_id)
+        except Organization.DoesNotExist:
+            return JsonResponse(
+                {"error": "Organization not found"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        headers = {
+            "Authorization": "Bearer {}".format(org.engage_token),
+            "Content-Type": "application/json",
+        }
+
+        result = requests.post(
```

Comment:
I'm not sure if you've seen this: https://github.com/praekeltfoundation/rp-sidekick/blob/develop/sidekick/utils.py#L27

We could apply some DRY here. 🙂

Reply:
Darn. I missed this. I'm going to have to refactor this a bit, but it will be better to it this way, thanks!